src/App.tsx

```diff
@@ -15,49 +16,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = '',
+  SORT_BY_ALPHABET = 'alphabet',
+  SORT_BY_LENGTH = 'length',
+}
+
+function sortedGoods(
+  goods: string[],
+  selectedOrder: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  if (selectedOrder) {
+    visibleGoods.sort((first: string, second: string) => {
+      switch (selectedOrder) {
+        case SortType.SORT_BY_ALPHABET:
+          return first.localeCompare(second);
+
+        case SortType.SORT_BY_LENGTH:
+          return first.length - second.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [selectedOrder, setSelectedOrder] = useState(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = sortedGoods(goodsFromServer, selectedOrder, isReversed);
+
+  const resetButton = () => {
+    setSelectedOrder(SortType.DEFAULT);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': selectedOrder !== SortType.SORT_BY_ALPHABET,
+          })}
+          onClick={() => setSelectedOrder(SortType.SORT_BY_ALPHABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': selectedOrder !== SortType.SORT_BY_LENGTH,
+          })}
+          onClick={() => setSelectedOrder(SortType.SORT_BY_LENGTH)}
         >
           Sort by length
         </button>
-
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
         </button>
+        {
+          (isReversed || selectedOrder) && (
```

Comment:
```suggestion
        {(isReversed || selectedOrder) && (
```
Would be great to create a variable for this condition